index.js

```diff
@@ -133,15 +133,26 @@ async function init() {
 
 	const {private: repoIsPrivate} = await fetchRepoInfo(`${user}/${repository}`);
 
-	const repoListingConfig = {
-		user,
-		repository,
-		ref,
-		directory: decodeURIComponent(dir),
-		token: localStorage.token,
-		getFullData: true,
-	};
-	let files = await listContent.viaTreesApi(repoListingConfig);
+	let repoListingConfig;
+	let files;
+	while (true) { // eslint-disable-line no-constant-condition
```

Comment:
The lint rule exists to avoid to ever risking an infinite loop. It looks like this already has an infinite loop if the path doesn't exist.

I'd probably suggest something like:

```js
const pathParts = decodeURIComponent(dir).split('/');
const repoListingConfig = {
 user,
 repository,
 token: localStorage.token,
 getFullData: true,
};

for (const [index, part] of pathParts.entries()) {
  if  (index === 0) {
    continue;
  }

  repoListingConfig.ref = pathParts.slice(0, index);
  repoListingConfig.directory = pathParts.slice(index);
  
  // `return await` because this whole snippet should be in 
  // a dedicated function that returns when successful or 
  // when the loop has finished
  return await listContent.viaTreesApi(repoListingConfig);
}
```

Reply:
Could you use my piece of code? I'm not a fan of:

- recreating the whole object every time
- split/slice/split every time
- calling `decodeURIComponent` every time (it should not decode multiple times)
- using `.split('/')[0]`